packages/linux-firmware/linux-firmware.spec

```diff
@@ -53,10 +49,471 @@ mkdir -p %{buildroot}/%{fwdir}/updates
 
 # Use zstd compression for firmware files to reduce size on disk. This relies on
 # kernel support through FW_LOADER_COMPRESS (and FW_LOADER_COMPRESS_ZSTD for kernels >=5.19)
-make DESTDIR=%{buildroot}/ FIRMWAREDIR=%{fwdir} install-zst
+install -d %{buildroot}/%{fwdir}
+./copy-firmware.sh --zstd --ignore-duplicates %{buildroot}/%{fwdir}
 
 %files
 %dir %{fwdir}
-%{fwdir}/*
-%license LICENCE.* LICENSE.* GPL* WHENCE
+%dir %{fwdir}/updates
+
+# Directory structures for firmware we DO want to include.
+# These directories are intentionally NOT excluded so we can include specific files from them.
+%dir %{fwdir}/bnx2x
+%dir %{fwdir}/cxgb4
+%dir %{fwdir}/cxgb4/configs
+%dir %{fwdir}/i915
+%dir %{fwdir}/intel
+%dir %{fwdir}/intel/ice
+%dir %{fwdir}/intel/ice/ddp
+%dir %{fwdir}/intel/ice/ddp-comms
+%dir %{fwdir}/intel/ice/ddp-wireless_edge
+%dir %{fwdir}/qed
+%dir %{fwdir}/tigon
+
+# ALLOWLIST APPROACH: Exclude unwanted firmware, then explicitly include what we need
+#
+# We use an allowlist (default-deny) approach for security and maintainability:
+# - New firmware added upstream is automatically excluded (safe default)
+# - We only ship exactly what we need
+# - Package size stays minimal and controlled
```

Comment:
By convention, includes come before excludes. It's usually more interesting to see what a package does contain rather than what it doesn't.